### PR TITLE
Bridge Pangolin <-> Millau

### PR DESCRIPTION
**Pull request by AurevoirXavier**
#621
#624

### CODE REVIEW

File: bin/node/cli/src/service.rs [hunk 1]
```diff
@@ -715,15 +741,26 @@ where
...
+			let timestamp = sp_timestamp::InherentDataProvider::from_system_time();
+			let slot =
+				sp_consensus_babe::inherents::InherentDataProvider::from_timestamp_and_duration(
+					*timestamp,
+					slot_duration,
+				);
+			let uncles =
+				sp_authorship::InherentDataProvider::<<Block as BlockT>::Header>::check_inherents();
+
+			Ok((timestamp, slot, uncles))
```
Comment (AsceticBear) [inline bin/node/cli/src/service.rs:762]: Compared with https://github.com/paritytech/polkadot/tree/master/node#L1156 `Ok((timestamp, slot))`

The return value should not contain `uncles`?

Comment (AurevoirXavier) [reply bin/node/cli/src/service.rs:762]: Because Polkadot doesn't require this `Inherents`

**Comment by AurevoirXavier** [comment]
DO NOT merge.

File: bin/node/runtime/pangolin/src/pallets/bridge_dispatch.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+pub use pallet_bridge_dispatch::Instance1 as WithMillauDispatch;
```
Comment (fewensa) [inline bin/node/runtime/pangolin/src/pallets/bridge_dispatch.rs:1]:
```suggestion
pub use pallet_bridge_dispatch::Instance1 as WithPangolinDispatchInstance;
```

Comment (fewensa) [reply bin/node/runtime/pangolin/src/pallets/bridge_dispatch.rs:1]: It's right, but can change use same style? `WithMillauDispatchInstance` `WithMillauGrandpaInstance` `WithMillauMessagesInstance`

Comment (AurevoirXavier) [reply bin/node/runtime/pangolin/src/pallets/bridge_dispatch.rs:1]: I don't think so.

Checkout these:
https://github.com/darwinia-network/darwinia-common/blob/master/bin/node/runtime/pangolin/src/pallets/collective.rs#L2
https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/src/lib.rs#L526

File: bin/node/runtime/pangolin/src/pallets/bridge_grandpa.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+pub use pallet_bridge_grandpa::Instance1 as MillauGrandpa;
```
Comment (fewensa) [inline bin/node/runtime/pangolin/src/pallets/bridge_grandpa.rs:1]:
```suggestion
pub use pallet_bridge_grandpa::Instance1 as WithMillauGrandpa;
```

same with bridge_dispatch (WithMillauDispatch) and bridge_messages (WithMillauGrandpa) or remove `With` in bridge_dispatch and bridge_messages

**Review by HackFisher — CHANGES_REQUESTED** [review]
Need to bump pangolin spec version if we are going to do upgrade.

**Review by HackFisher — APPROVED** [review]

**Pull request merged**